cmd/xl-v1-multipart-common.go

```diff
@@ -19,14 +19,13 @@ package cmd
 import (
 	"path"
 	"sync"
+	"time"
 )
 
-// writeUploadJSON - create `uploads.json` or update it with change
-// described in uCh.
-func (xl xlObjects) updateUploadJSON(bucket, object string, uCh uploadIDChange) error {
+// updateUploadJSON - add or remove upload ID info in all `uploads.json`.
+func (xl xlObjects) updateUploadJSON(bucket, object, uploadID string, initiated *time.Time, isRemove bool) error {
```

Comment:
I guess `time.Time{}` or `time.Unix(0,0)` creates an variable, but `nil` is not.  That's the difference.  However its all inside/masked by `addUploadID()` and `removeUploadID()` functions